app/controllers/questions_controller.rb

```diff
@@ -0,0 +1,75 @@
+class QuestionsController < ApplicationController
+  before_action :require_login, only: [:create, :destroy, :edit, :update]
+
+  def index
+    @questions = Question.all
+  end
+
+  def show
+    @question = Question.find(params[:id])
+    @answer = Answer.new
+    @comment = Comment.new
+    @vote = Vote.new
+  end
+
+  def new
+    @question = Question.new
+  end
+
+  def create
+    @question = Question.new(question_params)
+    tags = params[:tags].split(',')
+    if @question.save
+      tags.each do |tag|
+        check_tag = Tag.find_by(name: tag)
+        if check_tag
+          @question.tags << check_tag
+        else
+          @question.tags << Tag.create(name: tag)
+        end
+      end
+      redirect_to root_path
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @question = Question.find(params[:id])
+  end
+
+  def update
+    @question = Question.find(params[:id])
+    current_question_tags = @question.question_tags
+    current_question_tags.destroy_all
+    @question.assign_attributes(question_params)
+    tags = params[:tags].split(',')
+    if @question.save
+      tags.each do |tag|
+        check_tag = Tag.find_by(name: tag)
+        if check_tag
+          @question.tags << check_tag
+        else
+          @question.tags << Tag.create(name: tag)
+        end
+      end
+      redirect_to root_path
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    Question.find(params[:id]).destroy
+    redirect_to root_path
```

Comment:
Remember to check for success. If your app doesn't actually allow for deletion, remove this action